### PR TITLE
immediately show restore, set restore filename, cleanup old AS baked content on start

### PR DESCRIPTION
**Pull request by birarda**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7445/app_12481_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7445/HighFidelity-Beta-PR12481-d9d2b26519d611a3405bf3ae10043fa53ec30a60.exe

File: assignment-client/src/assets/AssetServer.cpp [hunk 1]
```diff
@@ -505,6 +506,38 @@ void AssetServer::cleanupUnmappedFiles() {
     }
 }
 
+void AssetServer::cleanupBakedFilesForDeletedAssets() {
+    qCInfo(asset_server) << "Performing baked asset cleanup for deleted assets";
+
+    std::set<AssetUtils::AssetHash> bakedHashes;
+
+    for (auto it : _fileMappings) {
```
Comment (Atlante45) [inline assignment-client/src/assets/AssetServer.cpp]: `const auto&`

File: assignment-client/src/assets/AssetServer.cpp [hunk 2]
```diff
@@ -505,6 +506,38 @@ void AssetServer::cleanupUnmappedFiles() {
...
+            AssetUtils::AssetHash hash = it.first.mid(AssetUtils::HIDDEN_BAKED_CONTENT_FOLDER.length(),
+                                                      AssetUtils::SHA256_HASH_HEX_LENGTH);
+
+            // add the hash to our set of hashes for which we have baked content
+            bakedHashes.insert(hash);
+        }
+    }
+
+    // enumerate the hashes for which we have baked content
+    for (auto hash : bakedHashes) {
```
Comment (Atlante45) [inline assignment-client/src/assets/AssetServer.cpp]: `const auto&`

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/7455/app_12481_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/7455/HighFidelity-Beta-PR12481-a0f7d4dd345dc122d2bd25fe27c635313424baf8.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/7455/HighFidelity-Beta-PR12481-a0f7d4dd345dc122d2bd25fe27c635313424baf8.dmg

**Pull request merged**